### PR TITLE
Add support for optional parameters, e.g. Database Name

### PR DESCRIPTION
**Pull request by abigperson**
When you create an RDS instance it is not required to specify a "Database Name" so if one was not created when the RDS instance was created a Custom Resource that references this instance/cluster is returning an error.

### CODE REVIEW

**Comment by RealSalmon** [comment]
Good catch!

I think we can limit this change to just the 'non-split' keys, as those with a '.' in the key are always going to be defined. Also, it should probably return None instead of 'na'.

Can you please also include a test specifically for an empty database name? I can throw this in if you aren't familiar with moto or don't have the time.

File: python/index.py [hunk 1]
```diff
@@ -39,9 +39,9 @@ def get_data_value(data_key, resource, data_map):
     resource_key = data_map[data_key]
     if '.' in resource_key:
         x, y = resource_key.split('.')
-        return resource[x][y]
+        return resource.get(x, {}).get(y, "na")
     else:
-        return resource[resource_key]
+        return resource.get(resource_key, "na")
```
Comment (RealSalmon) [inline python/index.py]: I think None would be a more appropriate default here

File: python/index.py [hunk 2]
```diff
@@ -39,9 +39,9 @@ def get_data_value(data_key, resource, data_map):
     resource_key = data_map[data_key]
     if '.' in resource_key:
         x, y = resource_key.split('.')
-        return resource[x][y]
+        return resource.get(x, {}).get(y, "na")
```
Comment (RealSalmon) [inline python/index.py]: I'm not sure this is needed. Those '.' keys are always going to be there, right?

Comment (abigperson) [reply python/index.py]: Yes, I agree

File: python/index.py (hunk 1 above)
Comment (abigperson) [reply python/index.py]: Yup... makes sense! I'll make the changes and add a test for this

**Comment by abigperson** [comment]
I just copied the test you were using and made a couple edits to test this... looks good

**Comment by RealSalmon** [comment]
many thanks!
